[docs] Add Nsight system docs on Ray (#40396)
ray-project/ray: doc/source/ray-core/handling-dependencies.rst

  - Non-existant variable example: ``{"ENV_VAR_NOT_EXIST": "${ENV_VAR_NOT_EXIST}:/home/admin/my_lib"}`` -> ``ENV_VAR_NOT_EXIST=":/home/admin/my_lib"``.

> - ``nsight`` (Union[str, Dict[str, str]]): Nsight profiler config which either (1) "default" value refering to the `default config <https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/nsight.py#L20>`_, (2) a dict representing Nsight profiler options and its' value.

Reviewer: ```suggestion
- ``nsight`` (Union[str, Dict[str, str]]): specifies the config for the Nsight profiler. The value is either (1) "default", which refers to the `default config <https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/nsight.py#L20>`_, or (2) a dict of Nsight profiler options and their values.
```

>   More detailed setup and usage can be found :ref:`here <profiling-nsight-profiler>`.

Reviewer: ```suggestion
  See :ref:`here <profiling-nsight-profiler>` for more details on setup and usage.
```
